src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -687,6 +734,13 @@ public virtual SelectExpandNode CreateSelectExpandNode(ResourceContext resourceC
         /// <param name="resourceContext">The context for the resource instance being written.</param>
         /// <returns>The created <see cref="ODataResource"/>.</returns>
         public virtual ODataResource CreateResource(SelectExpandNode selectExpandNode, ResourceContext resourceContext)
+        {
+            ODataResource resource = CreateResource(selectExpandNode, resourceContext, false) as ODataResource;
+            return resource;
+        }
+
+
+        private ODataResourceBase CreateResource(SelectExpandNode selectExpandNode, ResourceContext resourceContext, bool isDeletedResource)
```

Comment:
I think you should move this method to the section where other private methods are